tests/test_arithmetics.rs

```diff
@@ -292,44 +338,94 @@ impl PyNumberProtocol for LhsAndRhsArithmetic {
         format!("{:?} - RA", other)
     }
 
-    fn __rpow__(&self, other: &PyAny, _mod: Option<&'p PyAny>) -> String {
-        format!("{:?} ** RA", other)
+    fn __rmul__(&self, other: &PyAny) -> String {
+        format!("{:?} * RA", other)
     }
 
-    fn __add__(lhs: &PyAny, rhs: &PyAny) -> String {
-        format!("{:?} + {:?}", lhs, rhs)
+    fn __rlshift__(&self, other: &PyAny) -> String {
+        format!("{:?} << RA", other)
     }
 
-    fn __sub__(lhs: &PyAny, rhs: &PyAny) -> String {
-        format!("{:?} - {:?}", lhs, rhs)
+    fn __rrshift__(&self, other: &PyAny) -> String {
+        format!("{:?} >> RA", other)
     }
 
-    fn __pow__(lhs: &PyAny, rhs: &PyAny, _mod: Option<u32>) -> String {
-        format!("{:?} ** {:?}", lhs, rhs)
+    fn __rand__(&self, other: &PyAny) -> String {
+        format!("{:?} & RA", other)
+    }
+
+    fn __rxor__(&self, other: &PyAny) -> String {
+        format!("{:?} ^ RA", other)
+    }
+
+    fn __ror__(&self, other: &PyAny) -> String {
+        format!("{:?} | RA", other)
+    }
+
+    fn __rpow__(&self, other: &PyAny, _mod: Option<&'p PyAny>) -> String {
+        format!("{:?} ** RA", other)
+    }
+
+    fn __rmatmul__(&self, other: &PyAny) -> String {
+        format!("{:?} @ RA", other)
     }
 }
 
 #[pyproto]
-impl PyObjectProtocol for LhsAndRhsArithmetic {
+impl PyObjectProtocol for LhsAndRhs {
     fn __repr__(&self) -> &'static str {
         "BA"
```

Comment:
I wonder if `LhsAndRhs` should have a field `name: String`, and then this could be:

```suggestion
        &self.name
```

That way the tests could be written as e.g. `"assert a.__radd__(b) == 'B + A'"`

(I find `RA` and `BA` slightly confusing. It's easy enough to look back up at the pyclass to figure it out though, so this isn't a strong feeling.)

Reply:
I tried so but it wasn't so meaningful when we want to test which operator is called.